docs/cudf/source/developer_guide/testing.md

```diff
@@ -0,0 +1,150 @@
+# Testing cuDF
+
+## Tooling
+Tests in cuDF are written using [`pytest`](https://docs.pytest.org/en/latest/).
+Test coverage is measured using [`coverage.py`](https://coverage.readthedocs.io/en/latest/),
+specifically the [`pytest-cov`](https://github.com/pytest-dev/pytest-cov) plugin.
+Code coverage reports are uploaded to [Codecov](https://app.codecov.io/gh/rapidsai/cudf).
+Each PR also indicates whether it increases or decreases test coverage.
+
+## Test organization
```

Comment:
Is it worth mentioning that this is not how tests are _currently_ organized? Over time, we'll migrate our current tests to this new organization, but _new_ tests should follow these guidelines.

Reply:
This applies to a lot of things: tests, benchmarks, some of our code guidelines... I think what might make the most sense is to update the top-level https://docs.rapids.ai/api/cudf/stable/developer_guide/index.html with a note that everything in the guide is aspirational but not yet an accurate description of the current state of things. WDYT?

Reply:
Sure